packages/agreement-process/src/services/agreementService.ts

```diff
@@ -418,6 +422,40 @@ export function agreementServiceBuilder(
         )
       );
     },
+    async internalDeleteAgreementById(
```

Comment:
Since these routes are very specific for delegation revocation I would make the names (and also the routes) more specific
```suggestion
    async internalDeleteAgreementAfterDelegationRevocation(
```